### ISSUE TITLE
Auto-complete for accounts

### ISSUE THREAD

**Issue by aperomsik** [codinguser/gnucash-android#110]
I imported my account structure. I have way too many accounts in my desktop GNUcash file for selection from list to be practical. In desktop GNUcash I can usually type one or two characters and quickly find the account I am looking for.

**Comment by codinguser** [comment]
@MisterY you're right. This should probably stay open because it was not quite rejected, just low priority. 
The way I envision it, it would best be through subclassing AutocompleteTextBox to create something which searches as we want, suggests accounts as we want and also only accepts full account names. I would rather not have a separate view for picking account. That only adds the friction for entering transactions.

About the speed of entering transactions, most people have similar costs throughout the week. It is possible to start typing the description and auto-fill the rest of the transaction. 

That means the autocomplete of accounts would be great to have, but not a priority with the limited resources we have. 
I will however re-open this issue.

**Comment by MisterY** [comment]
I actually started some work on this, however slow. Do you mind checking Money Manager Ex for Android and let me know if picking up the account/category is too distracting? 
I find it quite convenient but would appreciate any comments before dedicating more time to this issue.

**Comment by MisterY** [comment]
I forgot to mention, and can't edit the post on mobile, that having a separate activity provides a lot of freedom in terms of functionality and allows for some nice search interface. This is my experience, at least.

**Comment by MisterY** [comment]
Unfortunately, the auto-fill does not very useful for my usage scenario. I tend to empty the database about once a month (and these days even more frequently as I'm testing various scenarios) and, every time I do that, the auto-fill needs to learn the transactions again.
Perhaps this is because I export the accounts into the new file (finally started the migration so the structure keeps changing) so the memorized transactions are lost. In any case, being able to type a few letters would help immensely.
I've created a couple of Python scripts for desktop. One of them does exactly this. :)) While migrating the data to GnuCash, I need to find specific accounts and adjust the transactions import via .qif. The search for full account name really helps.
